docs/maven-tutorial.md

```diff
@@ -0,0 +1,231 @@
+# OpenShift + Maven + Nexus Tutorial
+
+While developing your application with Java and Maven, you'll most likely be
+building many times. In order to shorten the build times of your
+pods, Maven dependencies can be cached in a local Nexus repository. This
+tutorial will step you through integrating a Nexus repository into your project.
+
+| Content: |
+|:----------------------------|
+|[Prerequisites](#prerequisites)|
+|[Setting up Nexus](#setting-up-nexus)|
+|[Modifying the Builder](#modifying-the-builder)|
+|[See Also](#see-also)|
+
+## Prerequisites
+
+This tutorial assumes that you'll be working with a project that is already set
+up for use with Maven. If you're interested in using Maven with your Java
+project, we highly recommend looking at
+[their guide](https://maven.apache.org/guides/getting-started/index.html).
+
+In addition, be sure to check your application's image for Maven mirror
+capabilities. Many images that use Maven have a 'MAVEN_MIRROR_URL' that you can
+use to simplify this process.
+
+## Setting up Nexus
+
+Download the official Nexus Docker image and use `new-app` to deploy it.
+```bash
+oc new-app sonatype/nexus
+```
+
+After the build has finished, run:
+```bash
+oc get svc
+```
+
+The output should resemble:
+```bash
+NAME      CLUSTER-IP       EXTERNAL-IP   PORT(S)    AGE
+nexus     172.30.170.249   <none>        8081/TCP   6s
+```
+
+Confirm that Nexus is running by navigating your browser to the new Cluster-IP
```

Comment:
You don't need to access it outside the cluster. The whole point is to
access it from builds running in the cluster.

Ben Parees | OpenShift

On Aug 28, 2016 11:05 PM, "Haoran Wang" <notifications@github.com> wrote:

> In docs/maven-tutorial.md
> <https://github.com/openshift/origin/pull/10645#discussion_r76548496>:
>
> > +```bash
> > +oc new-app sonatype/nexus
> > +```
> > +
> > +After the build has finished, run:
> > +```bash
> > +oc get svc
> > +```
> > +
> > +The output should resemble:
> > +```bash
> > +NAME      CLUSTER-IP       EXTERNAL-IP   PORT(S)    AGE
> > +nexus     172.30.170.249   <none>        8081/TCP   6s
> > +```
> > +
> > +Confirm that Nexus is running by navigating your browser to the new Cluster-IP
>
> Can we add steps the export the service , create a route ? since we cannot
> visit the service ip outside the cluster .
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/10645/files/3ce5141dc4f8cb321bde8d7564dcd1d513439510#r76548496>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEvl3sAmYLENCZb9CewOh6xjhOxGCX1Dks5qkkyVgaJpZM4JtP9->
> .
>